Formula/f/fsarchiver.rb

```diff
@@ -0,0 +1,61 @@
+class Fsarchiver < Formula
+  desc "File system archiver for linux"
+  homepage "http://www.fsarchiver.org"
+  url "https://github.com/fdupoux/fsarchiver/releases/download/0.8.7/fsarchiver-0.8.7.tar.gz"
+  sha256 "f7cd059e70fc57e888db282c622ec050c0dcdbaacc65e3c1eb163cd9d92d810d"
+  license "GPL-2.0-only"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+  depends_on "pkg-config" => :build
+
+  depends_on "bzip2"
+  depends_on "e2fsprogs"
+  depends_on "libgcrypt"
+  depends_on "lz4"
+  depends_on "lzo"
+  depends_on "xz"
+  depends_on "zlib"
+  depends_on "zstd"
+
+  on_macos do
+    depends_on "gettext"
+  end
+
+  on_linux do
+    depends_on "util-linux"
+  end
+
+  def install
+    args = [
+      "--prefix=#{prefix}",
+      "--sysconfdir=#{etc}",
+      "--disable-e2initrd-helper",
+      "MKDIR_P=mkdir -p",
+    ]
+    args += if OS.linux?
+      %w[
+        --enable-elf-shlibs
+        --disable-fsck
+        --disable-uuidd
+        --disable-libuuid
+        --disable-libblkid
+        --without-crond-dir
+      ]
+    else
+      ["--enable-bsd-shlibs"]
+    end
+
+    ENV["HOMEBREW_CCCFG"] = "sa"
+    system "autoreconf", "--force", "--install", "--verbose"
+    system "./configure", "--prefix=/usr", *args
+
+    system "/home/linuxbrew/.linuxbrew/bin/make"
+    system "/home/linuxbrew/.linuxbrew/bin/make", "install"
+  end
+
+  test do
+    system bin/"fsarchiver", "usage"
```

Comment:
Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Reply:
I do not know, most fsarchiver runs need root access and mess with disks.
I will see if there are any partitions which can be accessed without root access.

Reply:
Can we try to do something that requires root access while not actually having root access, and check that we get the expected failure message and exit code?